File: 1-easy/problems/0693-binary-number-with-alternating-bits/solution.ts
function hasAlternatingBits(n: any): boolean | number | string | any {
  """
  :type n: int
  :rtype: bool
  """
  n, curr = divmod(n, 2)
  while (n > 0) {
      if (curr == n % 2) {
          return false;
      n, curr = divmod(n, 2)
  return true;
}

export { hasAlternatingBits };

File: 1-easy/problems/0693-binary-number-with-alternating-bits/solution.py
class Solution:
    def hasAlternatingBits(self, n):
        """
        :type n: int
        :rtype: bool
        """
        n, curr = divmod(n, 2)
        while n > 0:
            if curr == n % 2:
                return False
            n, curr = divmod(n, 2)
        return True
